### PR TITLE
add .prettierrc

### PR DESCRIPTION
**Pull request by dorianmariefr**
makes it easier to run `prettier --write` (e.g. `pw` with my `dorian-pw` gem) on files to format them automatically

it would be nice to have the whole codebase formatted though

```json
{
  "semi": false,
  "rubySingleQuote": false
}
```

`semi: false` is for not putting semicolons at the end of lines in js

`rubySingleQuote: false` is for using double quotes, I would be fine with single quote also but prefer double quote so I don't have to think about which quote to use if I ever need interpolation

### CODE REVIEW

**Review by colby-swandale — APPROVED** [review]
this LGTM but I wish to have other core team opinions

**Comment by simi** [comment]
Isn't `rubocop -a` enough for Ruby? For JS there is no linting set.

**Comment by dorianmariefr** [comment]
rubocop does linting, prettier/plugin-ruby does formatting https://github.com/prettier/plugin-ruby (even if rubocop does a bit of formatting too)

**Comment by hsbt** [comment]
I have a same opinion with @simi .

**Comment by dorianmariefr** [comment]
for js linting, it's eslint https://github.com/eslint/eslint

**Comment by simi** [comment]
But no JS linting is set in the app and it would be actually pain to introduce, since the JS codebase is not consistent. That means there is no rule currently on using semicolons and it doesn't make any sense to add prettier to format this (in one or another way).

_Btw. I was thinking about revisiting JS codebase and move to https://github.com/rails/importmap-rails and some small webcomponent library like [lit](https://lit.dev). But I'm waiting for Rails 7 release to revisit that idea. If that happen, I'll introduce some kind of JS linting as well._

**Comment by sonalkr132** [comment]
it should be fine to add .prettierrc to add .gitignore, so that you can continue using it locally.

**Comment by deivid-rodriguez** [comment]
I don't have any issues with any path taken, but note that @dorianmariefr can already add `.prettierrc` to his global gitignore to continue using it.

**Comment by simi** [comment]
Yup, global ignore is definitely way to go here for local stuff.

**Comment by dorianmariefr** [comment]
my `.git/info/exclude`:

```
/Procfile.dev
/.prettierrc
/bin/dev
/js
/css
```

js is a symlink to app/assets/javascripts

css is a symlink to app/assets/stylesheets

anyway, no problem, I will keep that for me

**Pull request closed without merging**
